### PR TITLE
Fix for possibly incorrect mean pooling

### PR DESCRIPTION
**Pull request by ir2718**
Hi,

this is the fix for issue https://github.com/SeanLee97/AnglE/issues/101, (in case it truly is incorrect). To the best of my knowledge, the current code sums over the whole attention mask tensor, instead of keeping dimension zero intact, and summing only over the first dimension.

### CODE REVIEW

**Comment by SeanLee97** [comment]
Thank you very much for your PR! 

After reviewing the implementation, you are absolutely correct!
I overlooked setting `keepdim=True` when summing the `attention_mask,` leading to a global sum being used.
